### PR TITLE
crimson/os/seastore/lba_manager: cleanups for indirect lba mappings

### PR DESCRIPTION
**Pull request by xxhdx1985126**
Signed-off-by: Xuehan Xu <xuxuehan@qianxin.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [x] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

File: src/crimson/os/seastore/lba_manager/btree/btree_lba_manager.h [hunk 1]
```diff
@@ -57,6 +57,21 @@ class BtreeLBAMapping : public BtreeNodeMapping<laddr_t, paddr_t> {
 //
 // NOTE THAT, for direct BtreeLBAMappings, their intermediate_keys are the same as
 // their keys.
+//
+// Indirect BtreeLBAMapping can be further categrozied into HALF_INDIRECT and
+// FULL_INDIRECT.
+// 1. HALF_INDIRECT mappings: indirect mappings that don't have information from
+//    the intermediate mapping. Specifically, they don't have valid intermediate_base,
+//    intermediate_offset or intermediate_length.
```
Comment (cyx1231st) [inline src/crimson/os/seastore/lba_manager/btree/btree_lba_manager.h:65]: Similarly,

HALF_INDIRECT -> RAW/INCOMPLETE_INDIRECT
FULL_INDIRECT -> INDIRECT

1. RAW/INCOMPLETE_INDIRECT mappings: they don't have information from the intermediate mapping and should not be exposed outside LBAManager. Specifically, they don't have valid intermediate_base, intermediate_offset, intermediate_length or attached to a logical cached extent.
2. INDIRECT mappings: they have the complete information from intermediate mappings for LBAManager users.

File: src/crimson/os/seastore/lba_manager.h [hunk 2]
```diff
@@ -100,8 +100,7 @@ class LBAManager {
     Transaction &t,
     laddr_t hint,
     extent_len_t len,
-    laddr_t intermediate_key,
-    laddr_t intermediate_base) = 0;
+    laddr_t intermediate_key) = 0;
```
[the remark this answers is not in the document]
Comment (xxhdx1985126) [reply src/crimson/os/seastore/lba_manager.h:103]: Um, actually, it's used by [CLONE_RANGE](https://github.com/ceph/ceph/pull/57735), perhaps I should merge this PR into the clone_range PR.

[the remark this answers is not in the document]
Comment (cyx1231st) [reply src/crimson/os/seastore/lba_manager.h:103]: Yeah.

File: src/crimson/os/seastore/lba_manager/btree/btree_lba_manager.cc [hunk 3]
```diff
@@ -764,15 +764,16 @@ BtreeLBAManager::_update_mapping(
     cache,
     c,
     [f=std::move(f), c, addr, nextent](auto &btree) mutable {
-      return btree.lower_bound(
+      return btree.upper_bound_right(
 	c, addr
       ).si_then([&btree, f=std::move(f), c, addr, nextent](auto iter)
 		-> _update_mapping_ret {
-	if (iter.is_end() || iter.get_key() != addr) {
```
[the remark this answers is not in the document]
Comment (xxhdx1985126) [reply src/crimson/os/seastore/lba_manager/btree/btree_lba_manager.cc:771]: I've just modified the code according to your suggestions in https://github.com/ceph/ceph/pull/57735. Closing this PR:-)

**Pull request closed without merging**